### PR TITLE
Add systemd support

### PR DESCRIPTION
**Pull request by andy-s-clark**
* Add support for systemd on RHEL 7 / CentOS 7
* Move creation of pid, socket and log directories out of the startup scripts and into puppet.

### CODE REVIEW

File: templates/uwsgi_systemd.service.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+#
+# uWSGI Emperor process systemd service
+#
+[Unit]
+Description=uWSGI Emperor
+After=network.target
+
+[Service]
+Type=simple
+ExecStart=/usr/sbin/uwsgi --die-on-term --ini <%= @config_file %>
```
Comment (jarshwah) [inline templates/uwsgi_systemd.service.erb]: Should you parameterise the uwsgi binary location or just leave the path off? I'm just imagining other distros that use systemd may not install uwsgi into sbin.

**Comment by jarshwah** [comment]
Could you please modify the README `service_provider` section and the `service_provider` comments in the head of the init.pp file to mention systemd support? Should be good to go from there.

File: templates/uwsgi_systemd.service.erb (hunk 1 above)
Comment (andy-s-clark) [inline templates/uwsgi_systemd.service.erb]: I'll parameterize the binary path. On CentOS 7, the service failed to start without the full path.
Any preference for the name of the parameter? "app_binary_directory"?

Comment (jarshwah) [inline templates/uwsgi_systemd.service.erb]: I'd just go with something like "binary_location" and include the `uwsgi` component. Thoughts?

Comment (andy-s-clark) [inline templates/uwsgi_systemd.service.erb]: I used the name "binary_directory" in order to match the variables "app_directory", "log_directory", etc.

Comment (jarshwah) [inline templates/uwsgi_systemd.service.erb]: Right, that makes sense. Fine with me.

**Comment by jarshwah** [comment]
@andy-s-clark I'm unsure if I'm going to release this or not. It depends on the outcome of https://github.com/cwood/puppet-uwsgi/issues/7

My module isn't very well maintained and includes no tests, so if the fork is willing to take over, then I'd be happy for that to happen. I'd then ask you to target your patch at the fork.

**Pull request merged**